### PR TITLE
Filter registered ROMs by emulator profile supported file types

### LINKED ISSUE #22: [Feature] If platform is PS3, only add specific file as the rom
**Is your feature request related to a problem? Please describe.**
When installing PS3 games every file in the folder structure gets added as an installed file. Only PS3_GAME/USRDIR/eboot.bin is a valid launch file. This creates a multitude of entries that need removed to get the correct play action.

**Describe the solution you'd like**
If the platform is PS3 then only add the above mentioned reboot.bin as the rom file.

**Describe alternatives you've considered**
The only alternative I have been able to come up with is manually editing a game after install as to not have 30+ play actions..

### PR DESCRIPTION
**Pull request by ryanpf**
Fixes #22

### CODE REVIEW

File: Games/RomMInstallController.cs [hunk 1]
```diff
@@ -96,10 +96,30 @@ public override void Install(InstallActionArgs args)
...
+                        var searchPatterns = (supportedFileTypes != null || supportedFileTypes.Count == 0)
+                            ? new List<string> { "*" }
+                            : supportedFileTypes.Select(x => "*." + x);
+
                         // Add just the ROM files to the list
-                        string[] actualRomFiles = Directory.GetFiles(installDir, "*", SearchOption.AllDirectories)
-                            .Where(file => !file.EndsWith(".m3u", StringComparison.OrdinalIgnoreCase))
-                            .ToArray();
+                        string[] actualRomFiles = searchPatterns.SelectMany(
+                            searchPattern => Directory.GetFiles(installDir, searchPattern, SearchOption.AllDirectories)
```
Comment (gantoine) [inline Games/RomMInstallController.cs:121]: i think we still need to ignore `m3u` files in the case where they're in the ZIP

File: Games/RomMInstallController.cs [hunk 2]
```diff
@@ -96,10 +96,30 @@ public override void Install(InstallActionArgs args)
...
+                        {
+                            var builtInProfile = (info.Mapping.EmulatorProfile as BuiltInEmulatorProfile);
+                            supportedFileTypes = API.Instance.Emulation.Emulators
+                                .FirstOrDefault(e => e.Id == info.Mapping.Emulator.BuiltInConfigId)?
+                                .Profiles
+                                .FirstOrDefault(p => p.Name == builtInProfile.Name)?
+                                .ImageExtensions;
+                        }
+
+                        var searchPatterns = (supportedFileTypes != null || supportedFileTypes.Count == 0)
```
Comment (gantoine) [inline Games/RomMInstallController.cs:115]:
```suggestion
                        var searchPatterns = (supportedFileTypes == null || supportedFileTypes.Count == 0)
```

File: Games/RomMInstallController.cs [hunk 3]
```diff
@@ -96,10 +96,30 @@ public override void Install(InstallActionArgs args)
...
+                            var customProfile = info.Mapping.EmulatorProfile as CustomEmulatorProfile;
+                            supportedFileTypes = customProfile.ImageExtensions;
+                        }
+                        else if (info.Mapping.EmulatorProfile is BuiltInEmulatorProfile)
+                        {
+                            var builtInProfile = (info.Mapping.EmulatorProfile as BuiltInEmulatorProfile);
+                            supportedFileTypes = API.Instance.Emulation.Emulators
+                                .FirstOrDefault(e => e.Id == info.Mapping.Emulator.BuiltInConfigId)?
+                                .Profiles
+                                .FirstOrDefault(p => p.Name == builtInProfile.Name)?
```
Comment (gantoine) [inline Games/RomMInstallController.cs:111]: any chance either of the `FirstOrDefault` will be null?

**Review by gantoine — CHANGES_REQUESTED** [review]

File: Games/RomMInstallController.cs (hunk 2 above)
Comment (ryanpf) [reply Games/RomMInstallController.cs]: Nice catch!

File: Games/RomMInstallController.cs (hunk 1 above)
Comment (ryanpf) [reply Games/RomMInstallController.cs:121]: That won't interfere with my use case, but I'm curious: won't it look like a bug if it ignores `m3u` files when a user has specified them as supported files in the emulator configuration?

Comment (gantoine) [reply Games/RomMInstallController.cs:121]: yes in the case where the user has specified it, but in the one where `supportedFileTypes == null || supportedFileTypes.Count == 0` the search pattern `new List<string> { "*" }` should still ignore `.m3u` files to match the current behaviour.

Comment (ryanpf) [reply Games/RomMInstallController.cs:121]: Ah, got it! I will work on that

**Review by gantoine — APPROVED** [review]
looks great, thanks for the PR!!

**Pull request merged**